### ISSUE TITLE
Mempool requires dynamic fee txs to increase both tip and fee cap by price bump threshold

### ISSUE THREAD

**Issue by PlasmaPower** [ethereum/go-ethereum#23311]
The comment here https://github.com/ethereum/go-ethereum/blob/3526f690478482a02a152988f4d31074c176b136/core/tx_list.go#L296 says that increasing either the fee cap or the tip cap by the threshold is enough to accept the replacement. However, the code actually requires both be increased by the threshold, as either half of the if statement passing will lead to the return false, and the transaction being rejected for being an underpriced replacement.

I'm not sure if the comment or code is correct here, but one should definitely be updated so they're in sync. It seems to me like accepting a replacement tx which increases the tip by 15% but doesn't change the fee cap much (as the base fee is the largest part of the fee cap) makes sense, and the current code wouldn't accept many replacement transactions.

**Comment by PlasmaPower** [comment]
I just saw some discussion on this in the PR that introduced this code, https://github.com/ethereum/go-ethereum/pull/22898/files#r636583352 . It seems like the current behavior is the desired behavior, so the comment should be updated.

**Comment by rjl493456442** [comment]
Yes, it's the expected behavior, otherwise attackers can always reassign a higher feecap. 
Do you want to open a PR to fix the comment?

**Comment by EdNoepel** [comment]
If the user supplied both `maxFeePerGas` and `maxPriorityFeePerGas`, replacements should be allowed when only `maxPriorityFeePerGas` is bumped.  Requiring both to be bumped prevents use of the feecap as an arbitrarily high set-and-forget hard limit, harming the user experience.  I see how the bumping feecap could allow a spam attack, but not the tip.

**Comment by mining-visualizer** [comment]
100% agree with your comments @EdNoepel.  I think the current implementation is not well thought out.  It should be possible to increase the `maxPriorityFeePerGas` only.  Since doing so will directly cost the user more, spam attacks are not an issue.

**Comment by Sunset17** [comment]
Well can you please tell me how to get my Etherium back?  I was dumb about thinking I would get 1.5 Etherium back if I put 1.5 in but I did it and got scammed. I emailed etherscan and they caught it and returned it back to the chain. But I couldn’t get mine back that’s all I want. I have the email and chain. Sushi swap took it out there was 26 and 29 Etherium.
